### ISSUE TITLE
Accessibility and UI issues with typed table block

### ISSUE THREAD

**Issue by Scotchester** [wagtail/wagtail#7646]
Starting this issue to catalogue issues that remain in the [typed table block](https://docs.wagtail.io/en/latest/reference/contrib/typed_table_block.html) UI, especially accessibility-related ones.

I am already working on a few of these, denoted below.

**Note:** We also know we need to take a harder look at how the ARIA Grid pattern might serve this interface well, and using that may nullify some of the issues below.

## Keyboard navigation

- [ ] After creating a new column, focus should either move to the heading input of that column or back to the new column button (for easier insertion of multiple columns)
  - [ ] After creating a new row, focus should move to the first column input in that row, if the decision in the item above was to move to the heading input
- [ ] Should use ARIA Grid pattern

## Screen reader support

- [ ] After inserting a column, there is no indication of what type of column it is
- [ ] Should read the current row and column positions (e.g., "row 2, column 3") whenever changing cells or after inserting/removing rows/columns if that action changes the current position
- [ ] Column heading inputs need labels @Scotchester 
- [ ] Add column menu should have an `aria-expanded` update, defaulting to `false` and updating to `true` when open @Scotchester 

## General UI

- [x] Use SVG plus icons for add buttons for better visibility @Scotchester 
- [x] Add borders between rows to help know which row the delete button applies to
- [ ] Add missing bottom border in Safari https://github.com/wagtail/wagtail/pull/7670#pullrequestreview-853537536
- [ ] Confirmation of row/column deletion would be nice, to protect against accidental data loss
- [ ] Change plus icon on add column button to an x when clicked to indicate that you click that button again to close the add column menu

**Comment by mohammadareeb95** [comment]
@Scotchester I would like on work on some General UI fixes

**Comment by lb-** [comment]
@mohammadareeb95 I encourage to to read through the whole issue, all comments and all related PRs - see what gaps there are and maybe suggest something specific you can reproduce and give a go at fixing.

**Comment by thibaudcolas** [comment]
The change proposed in #7667 is ready for someone else to take on, with a few tweaks:

- The whole label needs to be set up for translation, _not_ concatenation of translateable text.
- We need to add the index of the column. Something like `Column heading for column {index}, {column type}`
